### PR TITLE
Expose 3 GPS Channel Unit Params, Add Basic Metric Support

### PR DESCRIPTION
**Pull request by stieg**
This large ish PR solves issue #734 by introducing support for basic metric values in our GPS system.  It does this by doing the following:

* Starting the foundation of a strong typed unit system and defining Length and Speed units
* Adjusting the JSON API to allow for the reading/writing of 3 GPS channels (altitude, distance, speed).
* Adjusting the internal method setter in our logger to respect the units decision as set by the API.

### CODE REVIEW

File: src/logger/loggerApi.c [hunk 1]
```diff
@@ -1237,42 +1238,82 @@ int api_getGpsConfig(struct Serial *serial, const jsmntok_t *json)
     json_int(serial, "alt", gpsCfg->altitude.sampleRate != SAMPLE_DISABLED, 1);
     json_int(serial, "sats", gpsCfg->satellites.sampleRate != SAMPLE_DISABLED, 1);
     json_int(serial, "qual", gpsCfg->quality.sampleRate != SAMPLE_DISABLED, 1);
-    json_int(serial, "dop", gpsCfg->DOP.sampleRate != SAMPLE_DISABLED, 0);
+    json_int(serial, "dop", gpsCfg->DOP.sampleRate != SAMPLE_DISABLED, 1);
+
+    json_objStartString(serial, "units");
+    json_string(serial, "altitude", gpsCfg->altitude.units, 1);
```
Comment (brentpicasso) [inline src/logger/loggerApi.c]: recommend making these distance and altitude keys consistent with the other keys in this message.

**Comment by brentpicasso** [comment]
It's readable and cohesive.  Only one suggestion on API naming.

File: src/logger/loggerApi.c (hunk 1 above)
Comment (stieg) [inline src/logger/loggerApi.c]: You mean use the string 'dist' and 'alt'?  Ok.

**Comment by brentpicasso** [comment]
:shipit

**Pull request merged**